Model/Api/CreateAcquiredCustomer.php

```diff
@@ -47,6 +47,9 @@ public function __construct(
      */
     public function execute($customerId = null): ?array
     {
+        // fixes issue where customerId is sometimes passed as null
+        $customerId = $customerId ?: $this->customerSession->getCustomer()->getId();
```

Comment:
Is there any possibility that the session may not have a customer ID? So we need to handle a potential exception here?

- [ ] Protect against `null` customer ID if necessary

Reply:
we don't create a acquired customer record if the customer id is not set, because the next if statement:

if ($this->customerSession->isLoggedIn()) {

will return false in that case.